imapclient/append.go

```diff
@@ -13,6 +13,8 @@ import (
 //
 // The options are optional.
 func (c *Client) Append(mailbox string, size int64, options *imap.AppendOptions) *AppendCommand {
+	// Keep caps in cache to avoid beginCommand to be stuck
+	c.Caps()
```

Comment:
The thing is that client.caps may be nil, which is why I added this call at first (this function acquires the mutex properly and populates client.caps if nil).

Reply:
`Literal` might get called in any other command as well, it's not limited to APPEND.

Reply:
Wich solution do you propose to don't be stuck?
The problem is that Append() calls beginCommand() that aqcuires the encoder mutex and then calls Literal() that is stuck because of calling again beginCommand() wich tries to acquire again the encoder mutex.

Reply:
I've suggested to stop callings `Caps` in `commandEncoder.Literal`, instead locking `c.mutex` and looking up `c.caps` directly.

Reply:
It's fine if the caps are nil.

Reply:
Right, I think we should automatically send a `CAPABILITY` command (but not wait for it to complete) after the greeting/authentication/STARTTLS, if the server didn't send the capabilities on its own.

Reply:
I don't think we should wait for capabilities though, as the cost for that is one roundtrip, ie. exactly the same as disabling `LITERAL-`.

Reply:
Does https://github.com/emersion/go-imap/pull/514 help with your server?

Reply:
Yes, the caps are populated correctly before calling Literal